### PR TITLE
Feature/last commit date

### LINKED ISSUE #6: [Feature Request]: Last Contribution Date In Certificate
### Preflight Checklist

- [X] I have read the Contributions.md for this project.
- [X] I agree to follow the Code of Conduct that this project adheres to.
- [X] I have searched the feature for a feature request that matches the one I want to file, without success.

### Problem Description

The certificate doesn't show when the last contribution was made by the user.

### Proposed Solution

Add the last contribution date to the certificate.

### Alternatives Considered

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by himanshusanecha**
# Description:

This pull request will fetch last commit date and add in the certificate.

Fixes #6

## Type of change:
-   [x] New feature (non-breaking change which adds functionality)
# Checklist:

-   [x] My code follows the style guidelines of this project.
-   [x] I have performed a self-review of my own code.
-   [x] My changes generate no new warnings.
-   [x] New and existing unit tests pass locally with my changes.

# Screenshots / Video:
Screenshot from 2022-03-03 13-30-57

### CODE REVIEW

File: src/controllers/certificate.controller.js [hunk 1]
```diff
@@ -33,9 +34,10 @@ exports.generateGithubCert = async (req, res) => {
         if (repo.source) {
             throw new Error("Certificate can't be generated for forks");
         }
-        const [commits, pullRequests] = await Promise.all([
+        const [commits, pullRequests, lastCommit] = await Promise.all([
```
Comment (Shubham-Kumar-2000) [inline src/controllers/certificate.controller.js]: You don't need  to call an extra API just for that. 

You can just extract the Latest commit form commits and latest PR from pullRequests then take the max of that.

File: src/views/certificate.ejs [hunk 2]
```diff
@@ -47,8 +47,10 @@
...
                         This is to certify that <br/>
                         <span style='color: #e0442c;font-weight: 900;font-size: 3.5rem;'> <%= data.userName%> </span><br/>
-                        has actively contibuted to open source project <b><%= data.projectRepo %></b> of <b><%= data.projectOwner %>.</b>
+                        has actively contibuted to open source project <b><%= data.projectRepo %></b> of <b><%= data.projectOwner %>.</b> 
+                        
                     </div>
+                    <h2><b>Last commit date: </b> <%= data.lastCommitDate %></h2>
                 </div>
             </center> 
             <div class="flex-container">
```
Comment (Shubham-Kumar-2000) [inline src/views/certificate.ejs:56]: As you have added latest commit. reduce the top margin of `flex-container` for proper alignment.

**Review by Shubham-Kumar-2000 — CHANGES_REQUESTED** [review]

File: src/controllers/certificate.controller.js (hunk 1 above)
Comment (himanshusanecha) [reply src/controllers/certificate.controller.js]: I thought of going with this idea first but in commits, there is no timestamp associated, therefore switched to the second method.

Comment (Shubham-Kumar-2000) [reply src/controllers/certificate.controller.js]: its there `commits[i].commit.committer.date`

Comment (himanshusanecha) [reply src/controllers/certificate.controller.js]: Changed @Shubham-Kumar-2000

**Review by Shubham-Kumar-2000 — APPROVED** [review]

**Pull request merged**